mailbox/api/src/main/java/org/apache/james/mailbox/MessageIdManager.java

```diff
@@ -45,7 +46,7 @@
 
     List<MessageResult> getMessages(Collection<MessageId> messageIds, FetchGroup minimal, MailboxSession mailboxSession) throws MailboxException;
 
-    default Flux<MessageResult> getMessagesReactive(Collection<MessageId> messageIds, FetchGroup minimal, MailboxSession mailboxSession) {
+    default Publisher<MessageResult> getMessagesReactive(Collection<MessageId> messageIds, FetchGroup minimal, MailboxSession mailboxSession) {
         try {
             return Flux.fromIterable(getMessages(messageIds, minimal, mailboxSession));
```

Comment:
I know you try to take advantage of the default here, but then you still should not use `Flux` in api ?

Reply:
As an implementation detail but not part of the exposed API